core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -96,22 +96,27 @@ private[spark] class AppClient(
     /**
      *  Register with all masters asynchronously and returns an array `Future`s for cancellation.
      */
-    private def tryRegisterAllMasters(): Array[JFuture[_]] = {
+    private def tryRegisterAllMasters(): Array[Option[JFuture[_]]] = {
```

Comment:
you don't have to change the method signature here if you just to a flatten within this method